### PR TITLE
KIALI-735 Add Bullet Chart in Route Rules in the Service Detail

### PR DESCRIPTION
**Pull request by aljesusg**
Add horizontal chart (Bullet Chart) into route rules view to show the weights.
Jira: [KIALI-735](https://issues.jboss.org/browse/KIALI-735)

**Require**: https://github.com/kiali/kiali-ui/pull/412

- Fix some errors with rowkeys

## Weight 70-50

@serenamarie125 @abonas I am still working but I will be happy if you give me some feedback ^^

- [x] Triying to enable Legend, the component fail .
- [x] Working with Range values in bullet, it seems this value don't stack well

### CODE REVIEW

**Comment by aljesusg** [comment]
I think that we can remove the ranges (grey colors) 
screenshot from 2018-06-06 12-18-00

**Comment by xeviknal** [comment]
@aljesusg Great job!

I know this PR is in WIP, but I would like to bring up a special scenario for reviews-test-v2 route rule. In that case the bar should appear full because when there is only one version Istio route 100% of the traffic to that version.

**Comment by aljesusg** [comment]
Changed @xeviknal 
screenshot from 2018-06-06 12-31-12

File: src/pages/ServiceDetails/ServiceInfo/ServiceInfoRouteRules/RouteRuleRoute.tsx [hunk 1]
```diff
@@ -74,6 +83,22 @@ class RouteRuleRoute extends React.Component<RouteRuleRouteProps> {
     }));
   }
 
+  bulletChartValues = () => {
+    return (this.props.route || []).map((routeItem, u) => ({
+      value: this.props.route.length === 1 ? 100 : routeItem.weight,
+      title: `${u}_${routeItem.weight}`,
+      color: PFBlueColors[u],
```
Comment (xeviknal) [inline src/pages/ServiceDetails/ServiceInfo/ServiceInfoRouteRules/RouteRuleRoute.tsx]: Since we have a ranged set of colors, this line would break in case we have more than 6 versions. I would protect that somehow. For example, repeating colors:

```
color: PFBlueColors[u%PFBlueColors.length]
```
using that, we will only use values from 0 to 5.

File: src/pages/ServiceDetails/ServiceInfo/ServiceInfoRouteRules/RouteRuleRoute.tsx [hunk 2]
```diff
@@ -139,9 +165,21 @@ class RouteRuleRoute extends React.Component<RouteRuleRouteProps> {
...
+        <div>
+          <BulletChart
+            id="bar-chart-1"
+            label="Load"
+            stacked={true}
+            thresholdWarning={100}
+            thresholdError={100}
+            details="Versions weight"
+            values={this.bulletChartValues()}
+            showLegend={true}
```
Comment (xeviknal) [inline src/pages/ServiceDetails/ServiceInfo/ServiceInfoRouteRules/RouteRuleRoute.tsx]: Is the legend working? Should it be set as false?

File: package.json [hunk 3]
```diff
@@ -18,8 +18,8 @@
     "deep-freeze": "0.0.1",
     "js-yaml": "3.11.0",
     "lodash": "4.17.10",
-    "patternfly": "3.45.3",
```
Comment (pilhuhn) [inline package.json]: I think the PF update should go into a separate PR.

File: src/pages/ServiceDetails/ServiceInfo/ServiceInfoRouteRules/RouteRuleRoute.tsx (hunk 2 above)
Comment (aljesusg) [reply src/pages/ServiceDetails/ServiceInfo/ServiceInfoRouteRules/RouteRuleRoute.tsx]: Yes, I was playing with this and I forgot remove it

File: package.json (hunk 3 above)
Comment (aljesusg) [reply package.json]: We use to do that in the backend I don't know why we don't do in the front, for me is fine, I should  change in https://github.com/kiali/kiali-ui/pull/407 too.

**Comment by aljesusg** [comment]
Fixed legend with ranges in https://github.com/patternfly/patternfly-react/pull/383

**Comment by abonas** [comment]
@serenamarie125 any feedback before merging?

**Comment by serenamarie125** [comment]
On PTO for about 5 more hours - don’t have the computer til then.  I like using the range at 100 rather than the red line (forget what that’s called in the component) thoughts?.  I can look further when I return

**Comment by serenamarie125** [comment]
@aljesusg I forgot to reply when returning from the airport.  @abonas thanks for re-pinging me.

I would suggest to NOT use the threshold, but use a range.  It would look like this 

screen shot 2018-06-13 at 9 19 19 am

screen shot 2018-06-13 at 9 21 21 am

screen shot 2018-06-13 at 9 21 01 am

What color palette are you using for the blues?

**Comment by aljesusg** [comment]
Hi @serenamarie125 
Thanks for reply.
About the color I am using.
PfColors.Blue, PfColors.Blue500, PfColors.Blue600,PfColors.Blue300,PfColors.Blue200,PfColors.Blue100

**Comment by aljesusg** [comment]
Update PR @serenamarie125 
## Weight 50-50


## Weight 70-50

**Comment by aljesusg** [comment]
@xeviknal go and merge it ^^ @serenamarie125  approved !!!

**Comment by aljesusg** [comment]
Thanks @xeviknal

**Comment by abonas** [comment]
:cake:
